servicetalk-http-api/src/main/java/io/servicetalk/http/api/PayloadInfo.java

```diff
@@ -23,6 +23,15 @@
  * {@link StreamingHttpResponse}.
  */
 interface PayloadInfo {
+    /**
+     * Returns {@code true} if and only if, the payload body {@link Publisher} was never assigned, changed, or modified,
+     * and therefore remains empty.
+     *
+     * @return {@code true} if and only if, the payload body {@link Publisher} was never assigned, changed, or modified
```

Comment:
What would be result for an explicitly empty payload such as empty String or empty byte array? 

I am thinking of the case were app has their own empty payload response and they use it as the payload for a response they intend to be empty.

Reply:
Unfortunately, in order to support [this](https://github.com/apple/servicetalk/pull/1479/files#diff-06e897b4cfb03b8c7b26e0a72492fae2bc8406a983f0a1e529afb0172027c529R56-R62) use-case, we can not apply the optimization in the scenarios you highlighted, because users can expand it later. If in future we will agree to drop support for this way, we will be able to optimize any empty payload body.

However, if users set a read-only empty buffer, we can support it. Fixed: https://github.com/apple/servicetalk/pull/1479/commits/69e962fce3da0d562daa6e6d4ffad8094ec12c55